security-analysis-ws/src/main/java/eu/itesla_project/security/rest/api/impl/SecurityAnalysisServiceImpl.java

```diff
@@ -0,0 +1,205 @@
+                }
+            if (contingenciesFilePath != null)
+                try {
+                    Files.delete(contingenciesFilePath);
+                } catch (IOException e) {
+                    LOGGER.warn("Error deleting " + contingenciesFilePath, e);
+                }
+        }
+    }
+
+    private StreamingOutput toStream(SecurityAnalysisResult result, LimitViolationFilter limitViolationFilter,
+            Format format) {
+        Objects.requireNonNull(format);
+        return new StreamingOutput() {
+            
+            @Override
+            public void write(OutputStream out) throws IOException, WebApplicationException {
+                Objects.requireNonNull(out);
+                if (format.equals(Format.JSON)) {
+                    SecurityAnalysisResultSerializer.write(result, limitViolationFilter, out);
+                } else {
+                    try( Writer wr = new OutputStreamWriter(out, StandardCharsets.UTF_8)){
+                        CsvTableFormatterFactory csvTableFormatterFactory = new CsvTableFormatterFactory();
+                        Security.printPreContingencyViolations(result, wr, csvTableFormatterFactory, limitViolationFilter);
+                        Security.printPostContingencyViolations(result, wr, csvTableFormatterFactory, limitViolationFilter);
+                    }
+                }
+            }
+        };
+    }
+
+    private String getParameter(List<InputPart> parts){
+        Objects.requireNonNull(parts);
+        try {
+            return parts.stream().findFirst().get().getBodyAsString();
+        } catch (IOException e) {
+            LOGGER.error("Error reading parameter",e);
+        }
+        return null;
+    }
+
+    private Path getFile(List<InputPart> parts) throws IOException {
+        Objects.requireNonNull(parts);
+
+        for (InputPart inputPart : parts) {
+            Optional<String> filename_header = Arrays
+                    .asList(inputPart.getHeaders().getFirst("Content-Disposition").split(";")).stream()
+                    .filter(n -> n.trim().startsWith("filename")).findFirst();
+            if (filename_header.isPresent()) {
+                String[] filename_tokens = filename_header.get().split("=");
+                if (filename_tokens.length > 1) {
+                    String filename = filename_tokens[1].trim().replaceAll("\"", "");
+                    if (!filename.equals("")) {
+                        InputStream istream = inputPart.getBody(InputStream.class, null);
+                        return saveFile(istream, filename);
+                    }
+                }
+            }
+        }
+        return null;
+    }
+
+    private Path saveFile(InputStream uploadedInputStream, String filename) throws IOException {
+        Objects.requireNonNull(uploadedInputStream);
+        Objects.requireNonNull(filename);
+        File f = new File(System.getProperty("java.io.tmpdir"), filename);
```

Comment:
This is not thread safe: two call using the same file name will cause weird behavior. Consider the WorkingDirectory class